telos-backend/src/routes/api/journal.js

```diff
@@ -1,14 +1,135 @@
+  const todos = await TodoTask.find({
+    $or: [
+      { createdDate: { $gte: date + 'T00:00:00', $lte: date + 'T23:59:59' } },
+      { dueDate: { $gte: date + 'T00:00:00', $lte: date + 'T23:59:59' } },
+      { dueDate: { $lt: date + 'T00:00:00' }, completed: false },
+    ],
+  });
+  return todos;
+}
+
+// Param: date, of type date, parsed from url path param. Given as yyyy-mm-dd
+async function getHabitsForDay(date) {
+  // Create habit objects to return. These are not stored directly (since technically could be unlimited)
+  // Must find all habits that are between the requested day, but before end date, then check if they fall
+  // on the current day.
+  const habits = await Habit.find({
+    startDate: { $lte: date + 'T23:59:59' },
+    endDate: { $gte: date + 'T00:00:00' },
+  });
+  // Habit objects returned in array as following:
+  // {
+  //    name: String
+  //    done: Boolean
+  // }
+  // Note: The Date.getDay() method returns an integer for day of week: 0 for sunday, 1 for monday etc.
+  const day = new Date(date).getDay();
+  const dayOfWeek = getStringDayOfWeek(day);
+  const habitDataToSend = [];
+  for (let i = 0; i < habits.length; i++) {
+    // Check if habit falls on day of week.
+    const compDates = habits[i].completedDates;
+
+    // We must cast to string because the compDates array has Strings in it!
+    // (try console.log this value without it)
+    const completedValue = compDates.includes(String(new Date(date)));
+    if (habits[i].daysOfWeek.includes(dayOfWeek)) {
+      habitDataToSend.push({
+        _id: habits[i]._id,
+        date: date,
+        name: habits[i].name,
+        completed: completedValue,
+      });
+    }
+  }
+  return habitDataToSend;
+}
+
+async function getCalendarDataForDay(date) {
+  const calendarevents = await CalendarEvent.find({
+    startTime: {
+      $gte: date + 'T00:00:00',
+      $lte: date + 'T23:59:59',
+    },
+  });
+  return calendarevents;
+}
+// ------------------------------------
+// Endpoints defined from this point on
+
 // get widgets which appear in the journal for a single date.
 // params:
 // 'date': journal date to get widgets for
-router.get('/:date', (req, res) => {
+router.get('/:date', async (req, res) => {
+  const { date } = req.params;
+  // Need to get all the data from widgets on that date.
+  // Then go through each of components, if exist, and get the data.
+  const widgets = await Widget.find({ date: date });
+
+  // Must find all text widgets based on the widget Ids retrieved.
+  const textWidgetIds = [];
+  for (let i = 0; i < widgets.length; i++) {
+    if (widgets[i].type === 'text') {
+      textWidgetIds.push(widgets[i]._id);
+    }
+  }
+
+  const texts = await Text.find({ widgetId: { $in: textWidgetIds } });
+  const calendarData = await getCalendarDataForDay(date);
+  const habitData = await getHabitsForDay(date);
+  const todoData = await getTodosForDay(date);
+
   res.json({
```

Comment:
I just noticed that there is no status code, maybe add in status(200) here.

Reply:
Oh great catch, thanks again for the review! Have added, and fixed the text api lint issues